// src/utils/api.js
const apiUrl = process.env.REACT_APP_API_URL;

async function signUp(username, password) {
  const response = await fetch(`${apiUrl}/signup`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ username, password }),
  });

  if (!response.ok) {
    const errorMessage = await response.text();
    throw new Error(errorMessage);
  }
}

async function signIn(username, password) {
  const response = await fetch(`${apiUrl}/signin`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ username, password }),
  });

  if (!response.ok) {
    const errorMessage = await response.text();
    throw new Error(errorMessage);
  }

  const { token } = await response.json();
  localStorage.setItem('token', token);
}

async function signOut() {
  localStorage.removeItem('token');
}

async function getUser() {
  const token = localStorage.getItem('token');
  if (!token) {
    return null;
  }

  const response = await fetch(`${apiUrl}/user`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  if (!response.ok) {
    localStorage.removeItem('token');
    return null;
  }

  const { username } = await response.json();
  return { username };
}

export { signUp, signIn, signOut, getUser };
